File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/SpawnPaintingPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.api.network.packet.Packet;

import io.netty.buffer.ByteBuf;

public final record SpawnPaintingPacket(int entityId, String title, int x, int y, int z, int direction)
        implements Packet.OUTGOING {

    @Override
    public void serialize(ByteBuf os) {
        writeVarInt(os, entityId);
        writeString(os, title);
        os.writeInt(x);
        os.writeInt(y);
        os.writeInt(z);
        os.writeInt(direction);
    }

    @Override
    public byte getId() {
        return 0x10;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/SpawnPaintingPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class SpawnPaintingPacket(
    val entityId: Int,
    val title: String,
    val x: Int,
    val y: Int,
    val z: Int,
    val direction: Int
) : Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeVarInt(entityId)
        os.writeString(title)
        os.writeInt(x, y, z, direction)
    }

    override val id: Byte
        get() = 0x10
}
